pkg/metrics/procs_windows.go

```diff
@@ -657,20 +661,23 @@ func (self *ProcsMonitor) Sample() (results sample.EventBatch, err error) {
 			return errProcs
 		}
 
-		var containerDecorator ProcessDecorator = nil
-		if self.containerSampler.Enabled() {
-			containerDecorator, err = self.containerSampler.NewDecorator()
-			if err != nil {
-				// ensure containerDecorator is set to nil if error
-				containerDecorator = nil
-				if id := containerIDFromNotRunningErr(err); id != "" {
-					if _, ok := containerNotRunningErrs[id]; !ok {
-						containerNotRunningErrs[id] = struct{}{}
+		var containerDecorators []ProcessDecorator
+		for _, containerSampler := range self.containerSamplers {
+			if containerSampler.Enabled() {
+				decorator, err := containerSampler.NewDecorator()
+				if err != nil {
```

Comment:
why are we adding when it's nil? I understand that before it was not a slice and maybe "it was convenient". But now? cannot it be easily removed?

Reply:
You are right, it can be removed.

Reply:
Fixed in last commit https://github.com/newrelic/infrastructure-agent/pull/1725/commits/337fedfde8783cd63ae68c419a492281d32a517a